.github/workflows/iroha2-dev.yml

```diff
@@ -12,84 +13,82 @@ env:
 # `ubuntu-latest` and not `[self-hosted, Linux]`.
 jobs:
   deploy:
+    if: github.event.pull_request.merged
     runs-on: ubuntu-latest
     container:
       image: 7272721/i2-ci:latest
     steps:
       - uses: actions/checkout@v2
       - uses: Swatinem/rust-cache@v1
       - name: set up buildx
-        uses: docker/setup-buildx-action@v1
+        uses: docker/setup-buildx-action@v2
         id: buildx
         with:
           install: true
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build and push docker image
-        uses: docker/build-push-action@v2
+          driver: docker
+
+      - name: Build iroha2:base image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:base
           context: .
-          push: true
-          tags: hyperledger/iroha2:dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-      - name: Build and push docker image (iroha_client_cli)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.base
+
+      - name: Build iroha2:build docker image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:build
           context: .
-          push: true
-          tags: hyperledger/iroha2:client-cli-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=iroha_client_cli
-      - name: Build and push docker image (kagami)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.build
+
+      - name: Build iroha2:dev docker image
+        run: |
+          docker build -f Dockerfile -t hyperledger/iroha2:dev .
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
```

Comment:
I guess `docker/build-push-action@v3` requires login as a preliminary step, even if it does not push 
https://github.com/docker/build-push-action#usage

Reply:
We can try to move `docker/login-action` up and set it before `docker/build-push-action@v3`. By the way, if the problem in that, I guess building from forks should has the same issue.